### PR TITLE
fix #12395. input group hover style error

### LINKED ISSUE #12395: 输入框组合 hover 时界面抖动
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.9.3

### Environment
WIN10 - Chome69 - React 16

### Reproduction link
[https://ant.design/components/input-cn/](https://ant.design/components/input-cn/)

### Steps to reproduce
鼠标滑过输入框组合元素

### What is expected?
不抖动

### What is actually happening?
抖动

inputgroup

### PR DESCRIPTION
**Pull request by hengkx**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for branch `feature`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

**Comment by hengkx** [comment]
#12395

**Comment by hengkx** [comment]
qq20180926-114807

File: components/input/style/mixin.less [hunk 1]
```diff
@@ -263,6 +263,16 @@
     display: block;
     .clearfix;
 
+    &-addon,
+    &-wrap,
+    > .@{inputClass} {
+
```
Comment (zombieJ) [inline components/input/style/mixin.less]: unnecessary empty line

Comment (hengkx) [reply components/input/style/mixin.less]: removed

File: components/input/style/mixin.less [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 // == when hoverd
 .hover(@color: @input-hover-border-color) {
-  border-color: ~`colorPalette("@{color}", 5)`;
+  border-color: ~`colorPalette("@{color}", 5)` !important;
```
Comment (zombieJ) [inline components/input/style/mixin.less]: 可否尽量不用 important?

**Comment by hengkx** [comment]
image

**Comment by zombieJ** [comment]
Seems hover has some issue:
kapture 2018-09-26 at 17 58 42

**Comment by hengkx** [comment]
@zombieJ 这个我看了 是因为`Cascader`  里面 `ant-cascader-picker-label` 影响的, 可以 `pointer-events: none;` 来解决，但是只兼容IE11,
image
image
我有个疑问，选中的值 为什么 不用`input` 直接显示 而需要另外用`span`来显示。我不清楚，这样有没有其它用意。
image

image

有什么好的建议吗？

**Comment by hengkx** [comment]
这个gif工具 不太 好用。改了下`Cascader` `input` style,直接使用`Cascader`也是存在问题的

**Comment by zombieJ** [comment]
印象里用 span 代替 input 展示是因为旧版 IE 有 bug，具体的记不清了。
我这边看应该没什么问题。@afc163 @yesmeck,  double confirm?

**Review by zombieJ — APPROVED** [review]

File: components/cascader/style/index.less [hunk 3]
```diff
@@ -13,7 +13,7 @@
     background-color: transparent !important;
     cursor: pointer;
     width: 100%;
-    position: static;
+    position: relative;
```
Comment (afc163) [inline components/cascader/style/index.less]: 这里吧 #10433 #16738 搞挂了。

**Pull request merged**